app/controllers/user_gift_requests_controller.rb

```diff
@@ -5,7 +5,7 @@ class UserGiftRequestsController < ApplicationController
   before_action :validate_user, only: [:pending]
 
   def cancel
-    gift_request.destroy
+    @gift_request.destroy
```

Comment:
@Dinuuu ahi lo corregí y ademas le puse @gift_request porque asi lo nombra el método check_ownernship:

  def check_ownership
    request_to_cancel = gift_request_by_organization
    @gift_request = request_to_cancel || gift_request
    return render status: :forbidden,
                  text: "You don't own the item" unless gift_request.present? &&
                                                        gift_request.user_id == current_user.id
  end